src/Sarif/Writers/SarifLogger.cs

```diff
@@ -241,14 +243,24 @@ public void AnalysisStopped(RuntimeConditions runtimeConditions)
 
         public void Log(IRule rule, Result result)
         {
+            if (rule.Id != result.RuleId)
+            {
+                //The rule id '{0}' specified by the result does not match the actual id of the rule '{1}'
+                string message = string.Format(SdkResources.ResultRuleIdDoesNotMatchRule,
+                    result.RuleId.ToString(),
+                    rule.Id.ToString());
```

Comment:
I don't ever call string.format(object), instead, I explicitly convert all arguments to string before making the call, so that reviewers can understand what culture-specific operations, if any, are performed. These ToString() calls are still a bug, though, as the 'Id' properties are already strings. :P